### PR TITLE
Add release workflow

### LINKED ISSUE #217: Consider creating versioned releases
Recently, we have come across a couple of situations where folks were using a much older version of the controller. 

Part of the problem is that dependabot will not submit updates to pseudo versions, see [dependabot/dependabot-core#2028](https://www.github.com/dependabot/dependabot-core/issues/2028).

Let's consider creating versioned releases for the controller so dependents can more easily track updates.

### PR DESCRIPTION
**Pull request by zregvart**
The main purpose of this is to create a tag so that releases are picked up by DependaBot. Versions are computed as:

    0.1.<number of commits in the repository>

Or continue from the latest tag's `major.minor` version.

Resolves #217

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      - name: Compute version
+        id: version
+        shell: bash
+        run: |
+          set -o errexit
+          set -o pipefail
+          set -o nounset
+
+          latest=$(git describe --tags --abbrev=0 2>/dev/null || echo 0.1.0)
+          echo "version=${latest%.*}.$(git rev-list --count HEAD)" >> "$GITHUB_ENV"
```
Comment (simonbaird) [inline .github/workflows/release.yaml:55]: Maybe:
```suggestion
          echo "version=${latest%.*}.$(git rev-list --count ${latest}..HEAD)" >> "$GITHUB_ENV"
```

**Review by simonbaird — APPROVED** [review]

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+      - name: Compute version
+        id: version
+        shell: bash
+        run: |
+          set -o errexit
+          set -o pipefail
+          set -o nounset
+
+          latest=$(git describe --tags $(git rev-list --tags --max-count=1) 2>/dev/null || echo 0.1.0)
```
Comment (zregvart) [inline .github/workflows/release.yaml:54]: This was `git describe --tags --abbrev=0` and caused an error:

```
fatal: No tags can describe '08b828de853f3b06db364a19da257fa5b175413c'.
```

So I changed it to `git describe --tags $(git rev-list --tags --max-count=1)`

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      timestamp: ${{ steps.timestamp.outputs.timestamp }}
+
+    steps:
+      - name: Git Info
+        id: head_sha
+        env:
+          GH_TOKEN: ${{ github.token }}
+          GH_COBRA: 1
+        run: |
+          echo head_sha=$(gh api /repos/zregvart/enterprise-contract-controller/git/matching-refs/heads/main --jq '.[0].object.sha') >> "$GITHUB_OUTPUT"
```
Comment (lcarva) [inline .github/workflows/release.yaml:29]: Is this right? I see a personal repo in the query. 

I'm also not sure why we want to echo `head_sha=<the-sha>`. Easier to debug? Why not just use the `steps.head_sha.outputs.head_sha`?

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      - name: Compute version
+        id: version
+        shell: bash
+        run: |
+          set -o errexit
+          set -o pipefail
+          set -o nounset
+
+          latest=$(git describe --tags $(git rev-list --tags --max-count=1) 2>/dev/null || echo 0.1.0)
+          echo "version="${latest%.*}".$(git rev-list --count "${latest}"..HEAD)" >> "$GITHUB_ENV"
```
Comment (lcarva) [inline .github/workflows/release.yaml:55]: When trying it locally, this fails:

```bash
🎄 latest=$(git describe --tags $(git rev-list --tags --max-count=1) 2>/dev/null || echo 0.1.0)
lucarval@fedora ~/src/enterprise-contract/enterprise-contract-controller 
🎄 echo $latest
0.1.0
🎄 version="${latest%.*}".$(git rev-list --count "${latest}"..HEAD)
fatal: ambiguous argument '0.1.0..HEAD': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```

Do we need to provide a default value? Or maybe we just need to set the initial tag manually.

File: .github/workflows/release.yaml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+      - name: Release
+        uses: softprops/action-gh-release@de2c0eb89ae2a093876385947365aca7b0e5f844 # v1
+        with:
+          prerelease: true
```
Comment (lcarva) [inline .github/workflows/release.yaml:60]: Is it really a prerelease?

File: .github/workflows/release.yaml [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+          name: Rolling release
+          body: Stable rolling release. Version can be determined by running `ec version`
```
Comment (lcarva) [inline .github/workflows/release.yaml:62]: Do we need to update body and name? Not sure if "rolling release" makes sense here. Also `ec version` definitely does not.

File: .github/workflows/release.yaml (hunk 3 above)
Comment (zregvart) [reply .github/workflows/release.yaml]: > Is this right? I see a personal repo in the query.

Thanks, this was mistakenly left from trying this out on my fork.

> I'm also not sure why we want to echo `head_sha=<the-sha>`. Easier to debug? Why not just use the `steps.head_sha.outputs.head_sha`?

See https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-output-parameter

File: .github/workflows/release.yaml (hunk 4 above)
Comment (zregvart) [reply .github/workflows/release.yaml]: Thanks, we can be a bit safer here. I made this change when I already had tags in my fork and missed this case.

File: .github/workflows/release.yaml (hunk 5 above)
Comment (zregvart) [reply .github/workflows/release.yaml]: Thanks, copied from ec-cli, doesn't need to be...

File: .github/workflows/release.yaml (hunk 6 above)
Comment (zregvart) [reply .github/workflows/release.yaml]: Thanks, also a copy-paste from ec-cli...

File: .github/workflows/release.yaml (hunk 3 above)
Comment (lcarva) [reply .github/workflows/release.yaml]: Thanks! Forgot about that.

**Review by lcarva — APPROVED** [review]

**Pull request merged**